### ISSUE TITLE
Vault agent documentation about tls auth unclear

### ISSUE THREAD

**Issue by drawks** [hashicorp/vault#9180]
**Describe the bug**
When using the vault-agent in cache mode clients connecting to the agent/cache cannot use tls/cert authentication provided by the client. It is obvious why this doesn't work when you really think about how tls/ssl works, however it would be nice if the documentation made this clearer.

Additionally I think it is surprising behavior that if you configure the agent's `vault` stanza with `client_key` and `client_cert` that any client connecting to the agent and issuing `vault login -method=cert` will have the agent's key material used to authenticate against the server's cert authentication endpoint regardless of what cert/key is actually provided by the client.

**To Reproduce**
1.  run vault agent configured in cache mode with no configured auto-auth and no `client_key` or `client_cert` provided in the `vault` stanza.
2. attempt to use a known good configured cert pair to login to the cert auth endpoint at the agent's configured tcp listener.
3. observe the server respond via the agent with a `400  * client certificate must be supplied`

**Expected behavior**
I expect this to fail as described, however the documentation could use a bit more explicit language around the expectations of cert authentication and agent. In the course of doing experiments with vault agent I found both the above behaviors quite surprising until I really considered the data flows.

**Environment:**
* Vault Server Version (retrieve with `vault status`): 1.3.4
* Vault CLI Version (retrieve with `vault version`): 1.3.4
* Server Operating System/Architecture: linux

**Comment by drawks** [comment]
As I play with this a little bit more it occurs to me that this is not just surprising but potentially a security issue. Imagine someone is using vault-agent along with `cert` authentication configured along with `auth_auth` and a properly configured sink. This same person might also enable `cache` with `use_auto_auth_token = false` and a `listener` on loopback in order to provide a local cacheing proxy for various arbitrary vault client access. This is a totally valid and at first glance would appear secure. However an unprivileged user can now get an authentication token via `VAULT_ADDR=http://localhost:8100 vault login -method=cert` as the agent will automatically provide the tls credentials on EVERY upstream vault request.

IMHO the appropriate fix would be to shift the `client_key` and `client_cert` parameters from the `vault` stanza and into the parameters of the `auto_auth` stanza's `cert` method configuration hash. Otherwise it is dangerously misleading to users that the configuration as such:

```
vault {
        address = "https://vault.internal.domain.com:8200"
	client_key = "/etc/vault-agent/vault-auth.key"
	client_cert = "/etc/vault-agent/vault-auth.cert"
}

cache {
        use_auto_auth_token = false
}

auto_auth {
	method "cert" {}
        sink "file" {
                config = {
                        path = "/var/run/application/private/vault.token"
                }
        }
}

listener "tcp" {
         address = "127.0.0.1:8100"
	 tls_disable = true
}
```

gives all appearances as to providing an un-authenticated cacheing proxy which can be trivially used to authenticate without providing credentials.

**Comment by drawks** [comment]
Can we get the core/agent and auth/cert and maybe other appropriate labels on this issue? Aside from making this issue surfaced a little better it may help get this documentation and/or behavioral fix prioritized to match the danger it presents as a falling knife in configuring vault agent.

**Comment by calvn** [comment]
Hey @drawks, thanks for the report and sorry for going silent on this one. We were working behind the scenes to get this addressed, and should be resolved by #9819 (available in 1.5.1 and above). Going to mark this as closed.
